sympy/discrete/convolution.py

```diff
@@ -344,3 +345,71 @@ def convolution_subset(a, b):
         c[mask] += expand_mul(a[smask] * b[mask^smask])
 
     return c
+
+
+#----------------------------------------------------------------------------#
+#                                                                            #
+#                              Covering Product                              #
+#                                                                            #
+#----------------------------------------------------------------------------#
+
+def covering_product(a, b):
+    """
+    Returns the covering product of given sequences.
```

Comment:
Maybe a blank line should be added after this. I don't know how important it is but PEP 257 says

> Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description. The summary line may be used by automatic indexing tools; it is important that it fits on one line and is separated from the rest of the docstring by a blank line. The summary line may be on the same line as the opening quotes or on the next line.